### PR TITLE
Implement SSO on marketing website

### PR DESCRIPTION
**Pull request by danoc**
@stewsmith – This isn't good to merge yet, but can you take a look at what I have so far?

Still needs more design work.

**How to test:** On staging, click on "Log in", click on "Use single-sign on," then enter an email address. It should send an HTTP request and you should receive a response back saying that either there is no account with that email address or SSO is not enabled for that account.

### CODE REVIEW

**Comment by danoc** [comment]
This still needs to handle form validation and AJAX responses, but @stewsmith, can you take a first pass?

File: website-guts/templates/components/modals/signin_modal.hbs [hunk 1]
```diff
@@ -21,6 +21,14 @@ api_domain: true
                   <span class="password-related error-hide error-right">{{tr 'Password is required'}}</span>
                   <input type="password" name="password" required>
               </label>
+
+              <!--
```
Comment (stewsmith) [inline website-guts/templates/components/modals/signin_modal.hbs]: What is this comment?

File: website-guts/assets/js/components/modals.js [hunk 2]
```diff
@@ -116,6 +116,7 @@ function storeModalState(modalType, modalOpen) {
 }
 
 window.optly.mrkt.modal.open = function(modalArgs) {
+
```
Comment (stewsmith) [inline website-guts/assets/js/components/modals.js]: why whitespace?

**Comment by stewsmith** [comment]
LGTM

**Pull request merged**